src/configuration-store/configuration-store.ts

```diff
@@ -21,9 +21,6 @@
  * implementation specific and handled upstream.
  */
 export interface IConfigurationStore<T> {
-  servingStore: ISyncStore<T>;
-  persistentStore: IAsyncStore<T> | null;
```

Comment:
these seemed unecessary for this interface to expose, so I removed, which allowed implementing classes to make these fields private (see below). Please let me know if there was a specific reason for this and I can revert

Reply:
I appreciate the improvements but please keep the PR focused to be able to trace any changes in behavior back to a narrow commit - happy to have this improvement landed/discussed on its own merits.

Reply:
Regarding the change - agreed; my thinking at first was to require implementing classes to have these two parameters but the way you changed it to makes more sense since the persistentStore especially is optional

Reply:
100%, will revert this portion

Reply:
this is an annoying habit I have of including unrelated changes in my PRs, gotta stop that 😂